crates/ruma-events/src/room/message/url_preview.rs

```diff
@@ -0,0 +1,320 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    file::EncryptedContent,
+    room::{OwnedMxcUri, UInt},
+};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+/// Encrypted Image Preview info
+pub struct EncryptedPreviewImage {
+    /// The target URI
+    pub url: OwnedMxcUri,
+    /// Encrypted PreviewImage data information
+    #[serde(flatten)]
+    pub encrypted_image: Box<EncryptedContent>,
+}
+
+/// UrlPreview Metadata about an image.
+/// modelled after [OpenGraph Image Properties](https://ogp.me/#structured)
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct PreviewImage {
+    /// MXC url to the image
+    #[serde(rename = "og:image", alias = "og:image:url", skip_serializing_if = "Option::is_none")]
+    pub url: Option<OwnedMxcUri>,
+
+    /// Encrypted PreviewImage data
+    #[serde(rename = "matrix:image:encryption", skip_serializing_if = "Option::is_none")]
+    pub encrypted_image: Option<EncryptedPreviewImage>,
+
+    /// The size of the image in bytes
+    #[serde(
+        rename = "matrix:image:size",
+        alias = "og:image:size",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub size: Option<UInt>,
+
+    /// The width of the image in pixels.
+    #[serde(rename = "og:image:width", skip_serializing_if = "Option::is_none")]
+    pub width: Option<UInt>,
+
+    /// The height of the image in pixels.
+    #[serde(rename = "og:image:height", skip_serializing_if = "Option::is_none")]
+    pub height: Option<UInt>,
+
+    /// The mime_type of the image
+    #[serde(rename = "og:image:type", skip_serializing_if = "Option::is_none")]
+    pub mimetype: Option<String>,
+}
+impl PreviewImage {
+    /// Whether this is effectively empty
+    pub fn is_none(&self) -> bool {
+        self.url.is_none() && self.encrypted_image.is_none()
+    }
+    /// Whether this has any preview image information
+    pub fn is_some(&self) -> bool {
+        self.url.is_some() || self.encrypted_image.is_some()
+    }
+    pub fn is_encrypted_image(&self) -> bool {
+        self.encrypted_image.is_some()
+    }
+}
+
+/// Preview Information for a URL matched in the message text, according to
+/// [MSC 4095](https://github.com/matrix-org/matrix-spec-proposals/pull/4095)
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct UrlPreview {
+    /// The url this was matching on
+    #[serde(alias = "matrix:matched_url")]
```

Comment:
It seems like that in the unstable version it doesn't use the `matrix:` prefix.

Reply:
which is what `alias` does here (this is not using `rename`): it just uses the field name to serialize and deserialize which is `matched_url`. alias only means "also deserialize from this name".

Reply:
Oh sorry I didn't notice that it was an alias, not a rename.